### PR TITLE
[Translation] [russian] -  data_race

### PR DESCRIPTION
**Pull request by alexshd**

### CODE REVIEW

File: _content/tour/rus/data_race.article [hunk 1]
```diff
@@ -1,28 +1,28 @@
-Data Races
-A data race is when two or more goroutines attempt to read and write to the same resource at the same time.
+Состояния гонки
```
Comment (rumyantseva) [inline _content/tour/rus/data_race.article:1]:
```suggestion
Гонка данных
```

"Состояние гонки" is called "Race condition", maybe it's not so important how exactly this article is called, but let's keep exactly the same terms.

File: _content/tour/rus/data_race.article [hunk 2]
```diff
@@ -174,25 +173,25 @@ switch between the two Goroutines at a bad time.
...
-by the time it completes the write operation. You are very lucky this is happening every
-time and you can see it. But normally a data race like this happens "randomly" and is
-impossible to know about until it’s too late. Luckily Go has a race detector to help
-find data races.
+После операции `modify` происходит переключение контекста. Три операции
+больше не являются непрерывными, и горутина 2 завершается тем, что ее локальное значение оказывается неверным
+к моменту завершения операции записи. Вам очень повезло, что это происходит каждый раз
+и вы это видите. Но обычно подобная гонка данных происходит "случайно",
+и узнать о ней невозможно, пока не станет слишком поздно. К счастью, в Go есть детектор гонок, который помогает
+обнаружить состояния гонки данных.
```
Comment (rumyantseva) [inline _content/tour/rus/data_race.article:181]:
```suggestion
обнаружить гонки данных.
```

File: _content/tour/rus/data_race.article [hunk 3]
```diff
@@ -174,25 +173,25 @@ switch between the two Goroutines at a bad time.
...
 
-** Race Detection
+** Обнаружение гонок
 
-There are several ways to engage the race detector. You can use it with the run, build
-and test command. If you use it with the build command, you have to remember to run the
-program. They say an instrumented binary can slow my program down by ~20%.
+Есть несколько способов задействовать детектор гонок. Вы можете использовать его с помощью команд `run`, `build`
+и командой `test`. Если вы используете его с командой build, вам нужно не забыть запустить
```
Comment (rumyantseva) [inline _content/tour/rus/data_race.article:186]:
```suggestion
Есть несколько способов задействовать детектор гонок. Вы можете использовать его с помощью команд `run`, `build`
и `test`. Если вы используете его с командой build, вам нужно не забыть запустить
```

File: _content/tour/rus/data_race.article [hunk 4]
```diff
@@ -219,14 +218,14 @@ binary and then running it.
     Final Counter: 2
     Found 1 data race(s)
 
-You can see a race was detected when running the program. This would happen with or
-without the log statement inserted. When a race is detected, the program panics and
-provides this trace. The trace shows where there was unsynchronized access to the
-same shared state where at least one access was a write.
+Вы можете видеть, что при запуске программы была обнаружена гонка. Это произошло как с
+без вставленного оператора лога. При обнаружении гонки программа впадает в панику и
```
Comment (rumyantseva) [inline _content/tour/rus/data_race.article:222]:
```suggestion
Вы можете видеть, что при запуске программы была обнаружена гонка. Это могло прозойти как с
вставкой выражения для логирования, так и без нее . При обнаружении гонки программа паникует и
```

File: _content/tour/rus/data_race.article [hunk 5]
```diff
@@ -238,21 +237,20 @@ line 38. Both Goroutines were created on line 36.
...
+в строке 39 также была бы гонкой данных, если бы код обращался к счетчику
+переменной. Операция плюс плюс - это операция чтения, модификации и записи, при которой операционная система может легко переключить контекст.
 
-So how can you fix the code to make sure that you remove the data race? There are two
-tools you can use, atomic instructions and mutexes.
+Как же исправить код, чтобы устранить гонку данных? Есть два
+инструмента, которые можно использовать: атомарные инструкции и мьютексы.
 
-** Atomics
+** Атомические инструкции
```
Comment (rumyantseva) [inline _content/tour/rus/data_race.article:247]: Let's add the term to the dictionary.

File: _content/tour/rus/data_race.article [hunk 6]
```diff
@@ -1,28 +1,28 @@
-Data Races
-A data race is when two or more goroutines attempt to read and write to the same resource at the same time.
+Гонки данных
```
Comment (rumyantseva) [inline _content/tour/rus/data_race.article:1]: Let's add the term to the dictionary.

File: _content/tour/rus/data_race.article [hunk 7]
```diff
@@ -148,9 +147,9 @@ If you run this program you no longer get the same result of 4, now you get the
     Final Counter: 2
 
 
-What is happening? You are running into a data race bug that did exist before, but
-wasn’t happening. The call to log is now causing the scheduler to make a context
-switch between the two Goroutines at a bad time.
+Что происходит? Вы столкнулись с ошибкой гонки данных, которая существовала раньше, но не была выевлена.
```
Comment (rumyantseva) [inline _content/tour/rus/data_race.article:150]:
```suggestion
Что происходит? Вы столкнулись с ошибкой гонки данных, которая существовала раньше, но не была выявлена.
```

File: _content/tour/rus/data_race.article [hunk 8]
```diff
@@ -148,9 +147,9 @@ If you run this program you no longer get the same result of 4, now you get the
...
+Вызов log теперь заставляет планировщик выполнять переключение контекста
+переключение контекста между двумя горутинами и делаеть это в неподходящее время.
```
Comment (rumyantseva) [inline _content/tour/rus/data_race.article:152]:
```suggestion
Вызов log теперь заставляет планировщик выполнять переключение контекста
между двумя горутинами и делать это в неподходящее время.
```

**Review by rumyantseva — COMMENTED** [review]
Not a full review yet, suggested consistency/typo fixes in the first part of the translation.

File: _content/tour/rus/data_race.article [hunk 9]
```diff
@@ -365,27 +363,23 @@ Goroutine at a time can execute that code.
...
 
-It’s important to remember the Lock creates back pressure, so the longer it takes to
-get from the Lock to the Unlock, the more chance of Goroutines waiting for their turn.
-If you forget to call Unlock, then all Goroutines waiting will deadlock. This is why it’s
-critical that the call to Lock and Unlock happen in the same function. Make sure I’m
-doing the bare minimum synchronization you need in the code block, but at least the
-minimum.
+Важно помнить, что блокировка создает обратное давление, поэтому чем больше времени требуется, чтобы
+от блокировки до разблокировки, тем больше шансов, что горутины дождутся своей очереди.
+Если вы забудете вызвать Unlock, то все ожидающие горутины окажутся в тупике. Вот почему
```
Comment (shalimski) [inline _content/tour/rus/data_race.article:378]:
```suggestion
Если вы забудете вызвать Unlock, то все ожидающие горутины окажутся заблокированы (deadlock). Вот почему
```

File: _content/tour/rus/data_race.article [hunk 10]
```diff
@@ -365,27 +363,23 @@ Goroutine at a time can execute that code.
...
+важно, чтобы вызов `Lock` и `Unlock` происходил в одной и той же функции. Убедитесь, что вы  делаете минимальную синхронизацию
```
Comment (shalimski) [inline _content/tour/rus/data_race.article:379]:
```suggestion
важно, чтобы вызов `Lock` и `Unlock` происходил в одной и той же функции. Убедитесь, что вы  делаете минимальную синхронизацию.
```

File: _content/tour/rus/data_race.article [hunk 11]
```diff
@@ -365,27 +363,23 @@ Goroutine at a time can execute that code.
...
 
-This is very bad code where someone is trying to get in and out of the Lock so quickly
-they actually lose the synchronization and the race detector can’t even discover the
-problem.
+Это очень плохой код, в котором кто-то пытается войти и выйти из `Lock` так быстро.
+что фактически теряет синхронизацию, а детектор гонок даже не может обнаружить проблему.
```
Comment (shalimski) [inline _content/tour/rus/data_race.article:382]:
```suggestion
Это очень плохой код, в котором кто-то пытается войти и выйти из `Lock` так быстро,
что фактически теряет синхронизацию, а детектор гонок даже не может обнаружить проблему.
```

File: _content/tour/rus/data_race.article [hunk 12]
```diff
@@ -423,20 +417,20 @@ problem.
...
-There is a second type of mutex called a read/write mutex. It allows me to separate
-the locks around reads and writes. This is important since reading data doesn’t pose
-a threat unless a Goroutine is attempting to write at the same time. So this type of
-mutex allows multiple Goroutines to read the same memory at the same time. As soon
-as a write lock is requested, the reads are no longer issued, the write takes place,
-the reads can start again.
+Существует второй тип мьютексов, называемый мьютексом чтения/записи. Он позволяет разделить
+блокировки на чтение и запись. Это важно, поскольку чтение данных не представляет
+угрозы, если только горутина не пытается писать в то же самое время. Таким образом, этот тип
+мьютекс позволяет нескольким горутинам читать одну и ту же память одновременно. Как только
```
Comment (shalimski) [inline _content/tour/rus/data_race.article:431]:
```suggestion
мьютекса позволяет нескольким горутинам читать одну и ту же память одновременно. Как только
```

File: _content/tour/rus/data_race.article [hunk 13]
```diff
@@ -423,20 +417,20 @@ problem.
...
+как только запрашивается блокировка записи, чтение больше не производится, происходит запись,
```
Comment (shalimski) [inline _content/tour/rus/data_race.article:432]:
```suggestion
запрашивается блокировка записи, чтение больше не производится, происходит запись,
```

File: _content/tour/rus/data_race.article [hunk 14]
```diff
@@ -492,14 +486,14 @@ the reads can start again.
         rwMutex.RUnlock()
     }
 
-You can see the use of a read/write mutex where there are 8 Goroutines reading the
-length of a slice within a 10 millisecond delay of each other, and 1 Goroutine waking
-up within 100 milliseconds to append a value (write) to the slice.
+Вы можете увидеть использование мьютекса для чтения/записи, когда 8 горутин читают длину
+длину фрагмента с задержкой в 10 миллисекунд друг от друга, и 1 горутина просыпается
+пробуждается в течение 100 миллисекунд, чтобы добавить значение (запись) к фрагменту.
```
Comment (shalimski) [inline _content/tour/rus/data_race.article:491]:
```suggestion
в течение 100 миллисекунд, чтобы добавить значение (запись) к слайсу.
```

**Review by shalimski — APPROVED** [review]

**Pull request merged**